### PR TITLE
[SMALLFIX] Switch to using NetAddress from thrift defined WorkerNetAddress

### PR DESCRIPTION
**Pull request by apc999**
Purpose of this PR is to use `NetAddress` which is a defined class, rather than thrift-derived class, as possible in the code. So the detail of RPC framework is abstracted away

### CODE REVIEW

File: servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
 public final class MasterWorkerInfo {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
   /** Worker's address */
-  private final WorkerNetAddress mWorkerAddress;
+  private final NetAddress mWorkerAddress;
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java]: mNetAddress?

File: servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java [hunk 2]
```diff
@@ -190,7 +190,7 @@ public synchronized WorkerInfo generateClientWorkerInfo() {
   /**
    * @return the worker's address
    */
-  public WorkerNetAddress getAddress() {
+  public NetAddress getAddress() {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java]: getWorkerAddress

File: servers/src/main/java/tachyon/worker/TachyonWorker.java [hunk 3]
```diff
@@ -74,7 +73,7 @@
   /** The address for the rpc server */
   private InetSocketAddress mWorkerAddress;
   /** Net address of this worker */
-  private WorkerNetAddress mWorkerNetAddress;
+  private NetAddress mWorkerNetAddress;
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/TachyonWorker.java]: mNetAddress

File: servers/src/main/java/tachyon/worker/TachyonWorker.java [hunk 4]
```diff
@@ -198,7 +197,7 @@ public BlockWorkerClientServiceHandler getBlockWorkerServiceHandler() {
    *
    * @return the worker's net address
    */
-  public WorkerNetAddress getWorkerNetAddress() {
+  public NetAddress getWorkerNetAddress() {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/TachyonWorker.java]: getNetAddress

File: servers/src/main/java/tachyon/worker/WorkerContext.java [hunk 5]
```diff
@@ -36,7 +35,7 @@ private WorkerContext() {} // to prevent initialization
   private static WorkerSource sWorkerSource = new WorkerSource();
 
   /** Net address of this worker */
-  private static WorkerNetAddress sWorkerNetAddress;
+  private static NetAddress sWorkerNetAddress;
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: sNetAddress

File: servers/src/main/java/tachyon/worker/WorkerContext.java [hunk 6]
```diff
@@ -59,18 +58,18 @@ public static WorkerSource getWorkerSource() {
   }
 
   /**
-   * @return {@link WorkerNetAddress} object of this worker
+   * @return {@link NetAddress} object of this worker
    */
-  public static WorkerNetAddress getWorkerNetAddress() {
+  public static NetAddress getWorkerNetAddress() {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: getNetAddress

File: servers/src/main/java/tachyon/worker/WorkerContext.java [hunk 7]
```diff
@@ -59,18 +58,18 @@ public static WorkerSource getWorkerSource() {
...
   }
 
   /**
    * Sets {@link NetAddress} object of this worker.
    *
-   * @param workerNetAddress {@link WorkerNetAddress} object of this worker
+   * @param workerNetAddress {@link NetAddress} object of this worker
    */
-  public static void setWorkerNetAddress(WorkerNetAddress workerNetAddress) {
+  public static void setWorkerNetAddress(NetAddress workerNetAddress) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: setNetAddress

Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: and netAddress for the variable name

File: servers/src/main/java/tachyon/worker/block/BlockWorker.java [hunk 8]
```diff
@@ -137,7 +137,7 @@ public BlockWorker() throws IOException {
    */
   @Override
   public void start() throws IOException {
-    WorkerNetAddress workerNetAddress;
+    NetAddress workerNetAddress;
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockWorker.java]: netAddress?

**Comment by jsimsa** [comment]
@apc999 thanks for the cleanup, left a few comments

File: servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java (hunk 2 above)
Comment (apc999) [inline servers/src/main/java/tachyon/master/block/meta/MasterWorkerInfo.java]: done

File: servers/src/main/java/tachyon/worker/TachyonWorker.java (hunk 3 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/TachyonWorker.java]: done

File: servers/src/main/java/tachyon/worker/TachyonWorker.java (hunk 4 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/TachyonWorker.java]: done

File: servers/src/main/java/tachyon/worker/WorkerContext.java (hunk 5 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: done

File: servers/src/main/java/tachyon/worker/WorkerContext.java (hunk 7 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/WorkerContext.java]: done

File: servers/src/main/java/tachyon/worker/block/BlockWorker.java (hunk 8 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/BlockWorker.java]: changed to `address`

**Comment by apc999** [comment]
@jsimsa , PR updated according to your comments. Could you take a look ? Thanks

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**